### ISSUE TITLE
Markdown mode fails to use `modeCfg.underscoresBreakWords` properly

### ISSUE THREAD

**Issue by gibiansky** [codemirror/codemirror5#2307]
Hey there!

I use CodeMirror via IPython, which supports using embedded LaTeX in Markdown cells. The following text causes buggy behaviour:

```
a_{This text is italicized
```

The Markdown mode has a property called `modeCfg.underscoresBreakWords` which indicates whether underscores in the middle of works cause italics or bolding after them. If this is set (which it is in IPython), something like `a_b` does not cause italics. However, `a_{` still does, because the condition is implemented like this:

``` javascript
if (!modeCfg.underscoresBreakWords) {
  if (ch === '_' && stream.peek() !== '_' && stream.match(/(\w)/, false)) {
    var prevPos = stream.pos - 2;
    if (prevPos >= 0) {
      var prevCh = stream.string.charAt(prevPos);
      if (prevCh !== '_' && prevCh.match(/(\w)/, false)) {
        ignoreUnderscore = true;
      }
    }
  }
}
```

Note that on line 2 of the above code block we require

``` javascript
stream.match(/(\w)/, false))
```

which does not hold if the next character is `{`.

This can be fixed simply by changing that second line to

``` javascript
if (ch === '_' && stream.peek() !== '_' && stream.match(/(\w|\{)/, false)) {
```

I'm not sure this is the right decision, of course, but given that I suspect a _lot_ of people will want to use LaTeX in their Markdown, this is something that should be addressed. (Perhaps via another `modeCfg` flag). I'm afraid I'm not sure how you'd like to handle this, so I did not make a Pull Request.

**Comment by krassowski** [comment]
We are tracking this for JupyterLab in https://github.com/jupyterlab/jupyterlab/issues/6774. In the ideal world the multiplexing mode would allow to get the information from the "outer" mode (e.g. gfm) and we could condition whether to open the "inner" mode using the information on the previous token type.

**Comment by marijnh** [comment]
There is no mention of `underscoresBreakWords` anywhere in the CodeMirror codebase, so I think that's just not a feature that exists, and this may not have anything to do with the wrapping of modes.

**Comment by krassowski** [comment]
This is correct, there is no `underscoresBreakWords` any more - it used to be here:

https://github.com/codemirror/CodeMirror/blob/ece5c232a43e9260b4c7393de5560ba2b60a4036/mode/gfm/gfm.js#L116

I was referring to the https://github.com/codemirror/CodeMirror/issues/2307#issuecomment-36150188 since I found this issue when searching for any prior requests to make the `multiplexingMode` more context-aware and it happened to discuss the exact issue I was trying to fix. As this comment and the previous one (https://github.com/codemirror/CodeMirror/issues/2307#issuecomment-36093610) referred to issue with nesting modes being applied incorrectly in naive `multiplexingMode` and asked if this is tracked in Jupyter project, and we now have a related issue I just wanted to leave a cross-reference for anyone searching in the future :)

**Comment by krassowski** [comment]
Also, if anyone is interested I proposed a workaround for lack of context handling by `multiplexingMode` in https://github.com/jupyterlab/jupyterlab/pull/10867 but this is just a band-aid for now - certainly we will need to revisit this on our side with migration to CodeMirror 6 :)
